extensions/hibernate-search-orm-elasticsearch/deployment/src/main/java/io/quarkus/hibernate/search/orm/elasticsearch/deployment/HibernateSearchElasticsearchProcessor.java

```diff
@@ -203,11 +211,37 @@ void setStaticConfig(RecorderContext recorderContext, HibernateSearchElasticsear
                                     // we cannot pass a config group to a recorder so passing the whole config
                                     recorder.createStaticInitListener(puName, buildTimeConfig,
                                             configuredPersistenceUnit.getBackendAndIndexNamesForSearchExtensions(),
+                                            rootAnnotationMappedClassNames,
                                             integrationStaticInitListeners))
                             .setXmlMappingRequired(xmlMappingRequired));
         }
     }
 
+    private static Set<String> collectRootAnnotationMappedClassNames(IndexView index) {
+        // Look for classes annotated with annotations meta-annotated with @RootMapping:
+        // those classes will have their annotations processed on every persistence unit.
+        // At the moment only @ProjectionConstructor is meta-annotated with @RootMapping.
+        Set<DotName> rootMappingAnnotationNames = new LinkedHashSet<>();
+        // This is built into Hibernate Search, which may not be part of the index.
+        rootMappingAnnotationNames.add(PROJECTION_CONSTRUCTOR);
+        // Users can theoretically declare their own root mapping annotations
+        // (replacements for @ProjectionConstructor, for example),
+        // so we need to consider those as well.
+        for (AnnotationInstance rootMappingAnnotationInstance : index.getAnnotations(ROOT_MAPPING)) {
+            rootMappingAnnotationNames.add(rootMappingAnnotationInstance.target().asClass().name());
+        }
```

Comment:
I wonder if it would make sense to have this on the search side, so we could add any other annotations in the future without a need to update quarkus code ?

Reply:
Well `@RootMapping` is a meta-annotation. So we *can* add any other annotation in the future, we just have to annotate it with `@RootMapping`.